libs/httpproxy/proxy-camel-servlet/src/main/java/org/codice/proxy/http/HttpProxyCamelHttpTransportServlet.java

```diff
@@ -194,28 +166,36 @@ protected void doService(HttpServletRequest oldRequest, HttpServletResponse resp
       consumer = resolve(request);
     }
 
-    if (consumer == null) {
-      log.debug("No consumer to service request {}", request);
-      response.sendError(HttpServletResponse.SC_NOT_FOUND);
-      return;
-    }
+    try {
+      if (consumer == null) {
+        log.debug("No consumer to service request {}", request);
+        response.sendError(HttpServletResponse.SC_NOT_FOUND);
+        return;
+      }
 
-    // are we suspended?
-    if (consumer.getEndpoint().isSuspended()) {
-      log.debug("Consumer suspended, cannot service request {}", request);
-      response.sendError(HttpServletResponse.SC_SERVICE_UNAVAILABLE);
-      return;
-    }
+      // are we suspended?
+      if (consumer.getEndpoint().isSuspended()) {
+        log.debug("Consumer suspended, cannot service request {}", request);
+        response.sendError(HttpServletResponse.SC_SERVICE_UNAVAILABLE);
+        return;
+      }
 
-    if (consumer.getEndpoint().getHttpMethodRestrict() != null
-        && !consumer.getEndpoint().getHttpMethodRestrict().equals(request.getMethod())) {
-      response.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
-      return;
-    }
+      if (consumer.getEndpoint().getHttpMethodRestrict() != null
+          && !consumer.getEndpoint().getHttpMethodRestrict().equals(request.getMethod())) {
+        response.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
+        return;
+      }
 
-    if ("TRACE".equals(request.getMethod()) && !consumer.getEndpoint().isTraceEnabled()) {
-      response.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
-      return;
+      if ("TRACE".equals(request.getMethod()) && !consumer.getEndpoint().isTraceEnabled()) {
+        response.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
+        return;
+      }
+    } catch (IOException e) {
+      LOG.warn("Could not send error due to: ", e);
+    } finally {
+      if (consumer == null) {
```

Comment:
❗️ Never a good idea to suddenly change the way a try/catch block like this exits in the finally. If the blocks ends up exiting because of runtime exceptions, for example, then those are swallowed. Same for errors ...

Reply:
Oddly enough I though I took this out. Thanks for catching it, I'll change it when I get in tomorrow